custom_components/hacs/base.py

```diff
@@ -1073,12 +1078,32 @@ async def async_update_downloaded_custom_repositories(self, _=None) -> None:
             return
         self.log.info("Starting recurring background task for downloaded custom repositories")
 
+        repositories_to_update = 0
+        repositories_updated = asyncio.Event()
+
+        async def update_repository(repository: HacsRepository) -> None:
+            """Update a repository"""
+            nonlocal repositories_to_update
+            await repository.update_repository(ignore_issues=True)
+            repositories_to_update -= 1
+            if not repositories_to_update:
+                repositories_updated.set()
+
         for repository in self.repositories.list_downloaded:
             if (
                 repository.data.category in self.common.categories
                 and not self.repositories.is_default(repository.data.id)
             ):
-                self.queue.add(repository.update_repository(ignore_issues=True))
+                repositories_to_update += 1
+                self.queue.add(update_repository(repository))
+
+        async def update_coordinators() -> None:
```

Comment:
As this is now moved to asyncio.Event, shouldn't this be called directly as it now waits for the event to be set anyway, instead of added to the queue?

Reply:
Replaced with a background task in cab9abf